zerver/actions/message_send.py

```diff
@@ -972,6 +975,41 @@ def do_send_messages(
                     )
                     send_request.automatic_new_visibility_policy = new_visibility_policy
 
+            # Set the visibility_policy of the users mentioned in the message
+            # to "FOLLOWED" if "automatically_follow_topics_where_mentioned" is "True".
+            user_mentions = (
+                (
+                    send_request.rendering_result.mentions_user_ids
+                    | send_request.stream_wildcard_mention_user_ids
+                    | send_request.topic_wildcard_mention_user_ids
+                )
+                & send_request.active_user_ids
+            ) - send_request.all_bot_user_ids
```

Comment:
This is a performance-sensitive code path. We should be very careful to avoid doing any new database queries if there are no mentions in the message.

Reply:
Fixed it in the recent commit.